src/mam/mod_mam_rdbms_arch_async.erl

```diff
@@ -86,10 +86,13 @@ multi_name(Name, Times) ->
 
 %%% flush callbacks
 flush(Acc, Extra = #{host_type := HostType, queue_length := MessageCount}) ->
-    {FlushTime, Result} = timer:tc(fun do_flush_pm/2, [Acc, Extra]),
-    mongoose_metrics:update(HostType, ?PER_MESSAGE_FLUSH_TIME, round(FlushTime / MessageCount)),
-    mongoose_metrics:update(HostType, ?FLUSH_TIME, FlushTime),
-    Result.
+    mongoose_instrument:span(mod_mam_pm_flushed, #{host_type => HostType},
+                             fun do_flush_pm/2, [Acc, Extra],
+                             fun(Time, _Result) ->
+                                     #{time => Time,
+                                       time_per_message => round(Time / MessageCount),
+                                       count => MessageCount}
```

Comment:
It might be. I just made the assumption of not removing anything that was there already, and I would like to do such changes separately. Of course it makes sense, and I thought about the same actually.

Reply:
Sounds good to me, I just hope not to bypass any of the things we've discovered here and there once we arrive to the moment of touching it all. Make sure to have notes about these points.

Reply:
We have follow-up stories for MAM already, so we can add this. But actually someone might use `time_per_message`, and it would be easier to switch the metric name than to also have to change the calculations? I think that "don't delete metrics - only rename or add" is still a good rule of thumb.